src/net/server/middleware/processors/edns.rs

```diff
@@ -0,0 +1,294 @@
+            }
+        }
+
+        ControlFlow::Continue(())
+    }
+
+    fn postprocess(
+        &self,
+        request: &Request<Message<RequestOctets>>,
+        response: &mut AdditionalBuilder<StreamTarget<Target>>,
+    ) {
+        // https://datatracker.ietf.org/doc/html/rfc7828#section-3.3.2
+        // 3.3.2.  Sending Responses
+        //   "A DNS server that receives a query sent using TCP transport that
+        //    includes an OPT RR (with or without the edns-tcp-keepalive
+        //    option) MAY include the edns-tcp-keepalive option in the
+        //    response to signal the expected idle timeout on a connection.
+        //    Servers MUST specify the TIMEOUT value that is currently
+        //    associated with the TCP session."
+        //
+        // https://datatracker.ietf.org/doc/html/rfc9210#section-4.2
+        // 4.2. Connection Management
+        //   "... DNS clients and servers SHOULD signal their timeout values
+        //   using the edns-tcp-keepalive EDNS(0) option [RFC7828]."
+        if let TransportSpecificContext::NonUdp(ctx) = request.transport() {
+            if let Ok(additional) = request.message().additional() {
+                let mut iter = additional.limit_to::<Opt<_>>();
+                if iter.next().is_some() {
+                    if let Some(idle_timeout) = ctx.idle_timeout {
+                        match IdleTimeout::try_from(idle_timeout) {
+                            Ok(timeout) => {
+                                // Request has an OPT RR and server idle
+                                // timeout is known: "Signal the timeout value
+                                // using the edns-tcp-keepalive EDNS(0) option
+                                // [RFC7828]".
+                                if let Err(err) =
+                                    add_edns_options(response, |builder| {
+                                        builder.push(&TcpKeepalive::new(
+                                            Some(timeout),
+                                        ))
+                                    })
+                                {
+                                    warn!("Cannot add RFC 7828 edns-tcp-keepalive option to response: {err}");
+                                }
+                            }
+
+                            Err(err) => {
+                                warn!("Cannot add RFC 7828 edns-tcp-keepalive option to response: invalid timeout: {err}");
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // https://www.rfc-editor.org/rfc/rfc6891.html#section-6.1.1
+        // 6.1.1: Basic Elements
+        // ...
+        // "If an OPT record is present in a received request, compliant
+        //  responders MUST include an OPT record in their respective
+        //  responses."
+        //
+        // TODO: What if anything should we do if we detect a request with an
```

Comment:
Ignore. A middleware processor cannot change the behavior of the service. And changing to SERVFAIL does not help.